### ISSUE TITLE
`no-new-array` confusing error message leading to `new-for-builtins`

### ISSUE THREAD

**Issue by Abdullah-Rady** [sindresorhus/eslint-plugin-unicorn#2442]
The eslint-plugin-unicorn has conflicting rules when trying to create a new array in JSX. Specifically, the rules unicorn/no-new-array and unicorn/new-for-builtins give contradictory advice, making it impossible to create a new array without an ESLint error.

When using new Array(repeat), I get an error from eslint/unicorn/no-new-array. If I remove the new keyword and just use Array(repeat), then I get an error from eslint/unicorn/new-for-builtins. This creates a confusing situation where neither approach is valid according to the current rules.

The rules unicorn/no-new-array and unicorn/new-for-builtins appear to be in conflict with each other.

```js
const repeat = 5;

const MyComponent = () => (
  <>
    {
    Array(repeat) // This triggers `unicorn/no-new-array`
      .fill(0)
      .map((_, i) => (
        <div key={i}>
          Repeated Content
          {' '}
          {i + 1}
        </div>
      ))
      }
  </>
);

// Changing to the following code triggers `unicorn/new-for-builtins`

const MyComponentFixed = () => (
  <>
    {new Array(repeat) // This triggers `unicorn/new-for-builtins`
      .fill(0)
      .map((_, i) => (
        <div key={i}>
          Repeated Content
          {' '}
          {i + 1}
        </div>
      ))}
  </>
);


```
**Expected Behavior**

There should be a clear, non-conflicting recommendation for creating arrays in this context.

**Environment**

    eslint-plugin-unicorn version: 55.0.0
    Node.js version: 22.3.0
    ESLint version: 8.57.0

**Comment by axetroy** [comment]
`unicorn/no-new-array` should provide two suggestions that should fix the conflict
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/891842d539d0d002119e9c476a050a12d3936a75/rules/no-new-array.js#L72-L81

It should be fixed to

```
Array.from({ length: repeat })
```

**Comment by fregante** [comment]
I think this is not an issue / big deal, I call these multi-step fixes as they deal with separate issues.

- `no-new-array`: `Array(x)` can be safely turned into `new Array(x)` without knowing what `x` is. 
- `new-for-builtins`: `new Array(x)` can only be turned into `Array.from({length: x})` if you know the type. If `x` is not a number, `new Array(x)` is equivalent to `[x]`, not `[,,,,]` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Array#array_constructor_with_a_single_parameter

And then, if you have a third plugin:

- [`array-func/from-map`](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2357): tells you that your code can be improved further, to `Array.from({length: repeat}, i => <YourJsx/>)`

**Comment by fregante** [comment]
Judging from the description of `no-new-array`, the rule is not about `new`, but specifically about the `Array` constructor. See:

> When using the Array constructor with one argument, it's not clear whether the argument is meant to be the length of the array or the only element.

This is purely a documentation issue, the error should not be:

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/891842d539d0d002119e9c476a050a12d3936a75/rules/no-new-array.js#L12

But something like:

- Array constructors are unclear in intent, use either [x] or Array.from({length: x})
